### PR TITLE
Init

### PR DESCRIPTION
**Pull request by claudio-ferraro**

### CODE REVIEW

File: config/brands.php [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+return [
+    'brand_attribute_code' => env('BRAND_ATTRIBUTE_CODE', 'brand')
```
Comment (royduin) [inline config/brands.php]:
```suggestion
    'brand_attribute_code' => 'brand',
```

The `env()` is for variables which could be different on environments. This setting wouldn't.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+``` bash
+php artisan vendor:publish --provider="Rapidez\StatamicBrands\ServiceProvider"
+```
+
+### Brand attribute code (optional)
+The default attribute code used for brands is `brand`, but it is also possible that another attribute is used for this. You can change this by adding the variable `BRAND_ATTRIBUTE_CODE` to the environment variables (`.env`). 
+For example:
+``` .env
+BRAND_ATTRIBUTE_CODE="manufacturer"
+```
```
Comment (royduin) [inline README.md]:
```suggestion

The default attribute code used for brands is `brand`, but it is also possible that another attribute is used for this. You can change this in `config/statamic-brands.php`.
```

File: src/ServiceProvider.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        $this->commands([
+            SyncBrandsCommand::class
+        ]);
+
+        return $this;
+    }
+
+    public function bootConfig(): self
+    {
+        $this->mergeConfigFrom(__DIR__.'/../config/brands.php', 'brands');
```
Comment (royduin) [inline src/ServiceProvider.php]:
```suggestion
        $this->mergeConfigFrom(__DIR__.'/../config/statamic-brands.php', 'statamic-brands');
```

File: src/ServiceProvider.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    public function bootPublishables(): self
+    {
+        $this->publishes([
+            __DIR__.'/../resources/blueprints/collections/brands/brands.yaml' => resource_path('blueprints/collections/brands/brands.yaml'),
+            __DIR__.'/../resources/content/collections/brands.yaml' => base_path('content/collections/brands.yaml')
+        ], 'rapidez-collections');
+
+        $this->publishes([
+            __DIR__.'/../config/brands.php' => config_path('brands.php'),
```
Comment (royduin) [inline src/ServiceProvider.php]:
```suggestion
            __DIR__.'/../config/statamic-brands.php' => config_path('statamic-brands.php'),
```

**Review by royduin — CHANGES_REQUESTED** [review]

File: src/Actions/CreateBrands.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    {
+        if ($brands->isEmpty()) {
+            return new Collection();
+        }
+
+        return $brands->filter(fn ($brand) => !blank($brand->name) && !blank($brand->brand_code))->map(fn ($brand) => [
+            'title' => $brand->title,
+            'brand_code' => $brand->brand_code,
+            'path' => $brand->brand_code
+        ])->each(fn ($brand) => Entry::updateOrCreate($brand, 'brands', 'brand_code', $storeCode));
```
Comment (indykoning) [inline src/Actions/CreateBrands.php]: My personal preference for code style on this is
```suggestion
        return $brands->filter(fn ($brand) => !blank($brand->name) && !blank($brand->brand_code))
            ->map(fn ($brand) => [
                'title' => $brand->title,
                'brand_code' => $brand->brand_code,
                'path' => $brand->brand_code
            ])
            ->each(fn ($brand) => Entry::updateOrCreate($brand, 'brands', 'brand_code', $storeCode));
```

File: src/Actions/DeleteBrands.php [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+class DeleteBrands
+{
+    public function deleteOldBrands(Collection $brandCodes, ?string $storeCode): void
+    {
+        if ($brandCodes->isEmpty()) {
+            return;
+        }
+
+        $deletedBrands = Entry::whereCollection('brands')->where('locale', $storeCode)->filter(fn ($deletedBrand) => !$brandCodes->contains($deletedBrand->get('brand_code')));
+        $deletedBrands->each(fn ($deletedBrand) => $deletedBrand->delete());
```
Comment (indykoning) [inline src/Actions/DeleteBrands.php]:
```suggestion
        $deletedBrands = Entry::whereCollection('brands')
            ->where('locale', $storeCode)
            ->filter(fn ($deletedBrand) => !$brandCodes->contains($deletedBrand->get('brand_code')))
            ->each(fn ($deletedBrand) => $deletedBrand->delete());
```

File: README.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+### Brand attribute code (optional)
+
+The default attribute code used for brands is `brand`, but it is also possible that another attribute is used for this. You can change this in `config/statamic-brands.php`.
+
+### Multisite
+When configuring your multisite for Statamic, it is important to add the stores to the sites section in the published file `content/collections/brands.yaml`, for example:
+
+```yaml
+title: Brands
+sites:
```
Comment (BobWez98) [inline README.md:31]: if brands get published, isn't this automaticly generated by statamic?

File: src/Events/BrandsImportedEvent.php [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace Rapidez\StatamicBrands\Events;
+
+use Illuminate\Foundation\Events\Dispatchable;
+
+class BrandsImportedEvent
```
Comment (BobWez98) [inline src/Events/BrandsImportedEvent.php:7]: Where does this event get used?

File: README.md (hunk 7 above)
Comment (claudio-ferraro) [reply README.md:31]: No I had to add it manually after publishing

File: src/Events/BrandsImportedEvent.php (hunk 8 above)
Comment (claudio-ferraro) [reply src/Events/BrandsImportedEvent.php:7]: After the import this is dispatched

https://github.com/rapidez/statamic-brands/pull/1/files/2185e9c8bff29c65c47073894c033b21f246025e#diff-79dc63c8e2631568230453cefe52c8e3cbe1845c22e68c754b921660db579caaR56

**Pull request merged**